fabric-common-tags-api-v1/src/datagen/java/net/fabricmc/fabric/impl/tag/common/datagen/generators/BiomeTagGenerator.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.tag.common.datagen.generators;
+
+import net.minecraft.tag.BiomeTags;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.BiomeKeys;
+
+import net.fabricmc.fabric.api.datagen.v1.FabricDataGenerator;
+import net.fabricmc.fabric.api.datagen.v1.provider.FabricTagProvider;
+import net.fabricmc.fabric.api.tag.v1.CommonBiomeTags;
+
+public class BiomeTagGenerator extends FabricTagProvider.DynamicRegistryTagProvider<Biome> {
+	public BiomeTagGenerator(FabricDataGenerator dataGenerator) {
+		super(dataGenerator, Registry.BIOME_KEY, "biomes", "Biome Tags");
```

Comment:
Since the registry key for the biome registry is `minecraft:root / minecraft:worldgen/biome`, the tag path should be `worldgen/biome`.

```suggestion
		super(dataGenerator, Registry.BIOME_KEY, "worldgen/biome", "Biome Tags");
```

Otherwise, tags will be written to `data/c/tags/biomes/<path>.json` rather than the correct `data/c/tags/worldgen/biome/<path>.json`. Although separate from this pull request, Fabric API should support a constructor that specifies the path from the registry key.